File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09 - While Loop Exercise/01. Old Books .py
searched_book = input()
number_of_books = int(input())

counter = 0
is_found = False

current_book = input()

while counter < number_of_books:
    if current_book == searched_book:
        is_found = True
        break
    counter += 1
    if counter == number_of_books:
        break
    current_book = input()
if is_found:
    print(f"You checked {counter} books and found it.")
else:
    print("The book you search is not here!")
    print(f"You checked {counter} books.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/01. Old Books.js
function oldBooks(data) {
    let searched = data.shift();
    let books = Number(data.shift());
    
    let book = 1;
    let found = false;
    while (book <= books) {
        if (searched === data.shift()) {
            console.log(`You checked ${book - 1} books and found it.`);
            found = true
            break;
        }
        book += 1;
    }
    if (!found) {
        console.log(`The book you search is not here!`);
        console.log(`You checked ${books} books.`);
    }
}


oldBooks(['Troy', '8', 'Stronger', 'Life Style', 'Troy']);
// Expected Output:
// You checked 2 books and found it.

oldBooks(['The Spot', '4', 'Hunger Games', 'Harry Potter', 'Torronto', 'Spotify']);
// Expected Output:
// The book you search is not here!
// You checked 4 books.
